pydantic/fields.py

```diff
@@ -356,7 +356,7 @@ def include_in_schema(self) -> bool:
         """
         False if this is a simple field just allowing None as used in Unions/Optional.
         """
-        return len(self.validators) != 1 or self.validators[0][1] != is_none_validator
+        return self.type_ != NoneType
```

Comment:
@samuelcolvin any specific reason for this to test validators size and function instead of `NoneType` directly?

I.e. any reason why I should revert it back and test `NoneType` *additionally*, apart from the previous test?

Reply:
Both times I changed it I added tests for that case.

If all the tests pass you should be safe to change it.

Reply:
Then we're safe.